onnxruntime/core/xnnpack/optimizer/xnnpack_transformer.cc

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "core/xnnpack/optimizer/xnnpack_transformer.h"
+
+#include <deque>
+
+#include "core/framework/op_node_proto_helper.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/nhwc_transformer.h"
+#include "core/optimizer/selectors_actions/helpers.h"
+#include "core/optimizer/transpose_optimizer/optimizer_utils.h"
+#include "core/optimizer/utils.h"
+#include "core/providers/cpu/nn/pool_attributes.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+using namespace onnx_layout_transformation;
+
+namespace onnxruntime {
+
+Status XNNPackTransformer::ApplyImpl(Graph& main_graph, bool& modified, int /* graph_level */,
```

Comment:
I don't see much in the way of new unit tests for all the xnnpack related code. What is the code coverage percentage from the existing tests?

Reply:
The existing Conv C++ unit tests work.
All existing model tests passed, except ONNX Conv node tests whose weights are not constant so they are auto-skipped. 
I also wrote some other tests but they are not in this PR because they will need more discussion.